tests/cli/test_start_orion.py

```diff
@@ -0,0 +1,130 @@
+import re
+import signal
+import subprocess
+import sys
+import time
+
+import pytest
+
+ORION_START_ARGS = [
+    "prefect",
+    "orion",
+    "start",
+    "--host",
+    "0.0.0.0",
+    "--port",
+    "4200",
+    "--log-level",
+    "INFO",
+]
+SHUTDOWN_TIMEOUT = 10
+
+
+class TestUvicornSignalForwarding:
+    @pytest.mark.skipif(
+        sys.platform == "win32",
+        reason="SIGTERM is only used in non-Windows environments",
+    )
+    def test_sigint_sends_sigterm_then_sigkill(self):
+        with subprocess.Popen(
+            ORION_START_ARGS,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        ) as proc:
+            out = []
+            for line in proc.stdout:
+                out.append(line)
+                if b"Uvicorn running" in line:
```

Comment:
thanks for the heads up, I'm sorry about that. 

these tests should amount to a minute or so at most, depending on startup and shutdown time of the orion process.

just to be sure: were talking about tests with postgres and sqlite?

my first suspicion is that the orion subprocess is started too early (for the db), and somehow makes the first test hang while looking for `Uvicorn running` in stdout :thinking:

will have a look tomorrow. any additional thoughts are appreciated!

Reply:
e.g. maybe this is not the right way to get orion to talk to db and have a successful startup?

or maybe there is a better way to 'wait for ready'?

Reply:
might it be related to [this fixture](https://github.com/PrefectHQ/prefect/blob/1f386ea1b5b200af18f521616d95c76f840a5039/tests/fixtures/database.py#L45)?

Reply:
also, on my fork these tests [get skipped in the sqlite/postgres CI](https://github.com/ddelange/prefect/actions/runs/3909012432/jobs/6679672912#step:12:425). how are they triggering timeout then 🤔
```
SKIPPED [1] tests/cli/test_start_orion.py:24: Only running tests for services. This test does not require a service.
SKIPPED [1] tests/cli/test_start_orion.py:53: Only running tests for services. This test does not require a service.
SKIPPED [1] tests/cli/test_start_orion.py:82: Only running tests for services. This test does not require a service.
```

Reply:
Those service tests are fixed on `main` — those test runs are just failing though.

The timed out tests are at (for example) https://github.com/ddelange/prefect/actions/runs/3909012432/jobs/6679672997

Reply:
We do something similar at https://github.com/PrefectHQ/prefect/blob/main/src/prefect/testing/fixtures.py#L36 if that's helpful

Reply:
thanks, will have a look next week how to adapt.

protip: you can press `y` before copying the link from the url bar, to get a permalink to the selected lines

Reply:
adapted: https://github.com/ddelange/prefect/pull/14/files

but not merged into here yet, because its a bit flaky. it consistently [hangs](https://github.com/ddelange/prefect/actions/runs/4015674559/jobs/6897694561#step:13:160) on a call to re.search in a percentage of the runs.

i could replace re.search, but it would look a bit ugly to otherwise assert the order of signals. what do you think?

Reply:
merged